src/pydvl/valuation/methods/msr_banzhaf.py

```diff
@@ -0,0 +1,186 @@
+            indices=data.indices,
+            data_names=data.data_names,
+            algorithm=self.algorithm_name,
+        )
+
+        ensure_backend_has_generator_return()
+
+        self.utility.training_data = data
+
+        strategy = self.sampler.make_strategy(self.utility, self.coefficient)
+        processor = delayed(strategy.process)
+
+        with Parallel(return_as="generator_unordered") as parallel:
+            with make_parallel_flag() as flag:
+                delayed_evals = parallel(
+                    processor(batch=list(batch), is_interrupted=flag)
+                    for batch in self.sampler.generate_batches(data.indices)
+                )
+                for batch in Progress(delayed_evals, self.is_done, **self.tqdm_args):
+                    for evaluation in batch:
+                        if evaluation.kind == ValueUpdateKind.POSITVE:
+                            self._pos_result.update(evaluation.idx, evaluation.update)
+                        elif evaluation.kind == ValueUpdateKind.NEGATIVE:
+                            self._neg_result.update(evaluation.idx, evaluation.update)
+                        else:
+                            raise ValueError(
+                                "Invalid ValueUpdateKind: {evaluation.kind}"
+                            )
+
+                        self.result = _combine_results(
+                            self._pos_result, self._neg_result, data=data
+                        )
+
+                        if self.is_done(self.result):
+                            flag.set()
+                            self.sampler.interrupt()
+                            break
+
+                    if self.is_done(self.result):
+                        break
+
+        return self
+
+
+def _combine_results(
+    pos_result: ValuationResult, neg_result: ValuationResult, data: Dataset
+) -> ValuationResult:
+    """Combine the positive and negative running means into a final result.
+
+    We cannot simply subtract the negative result from the positive result because
+    this would lead to wrong variance estimates, misleading update counts and even
+    wrong values if no further precaution is taken.
+
+    TODO: Verify that the two running means are statistically independent (which is
+    assumed in the aggregation of variances).
+
+    Args:
+        pos_result: The result of the positive updates.
+        neg_result: The result of the negative updates.
+        data: The dataset used for the valuation. Used for indices and names.
+
+    Returns:
+        The combined valuation result.
+
+    """
+    # set counts to the minimum of the two; This enables us to ensure via stopping
+    # criteria that both running means have a minimal number of updates
+    counts = np.minimum(pos_result.counts, neg_result.counts)
+
+    values = pos_result.values - neg_result.values
+    values[counts == 0] = np.nan
+
+    variances = pos_result.variances + neg_result.variances
```

Comment:
Besides assuming independence, this was only valid under my wrong assumption that `variances` are sampling variances that are already divided by the `counts` of each running mean. 

I think in our current approach of working with sample variances and then dividing them by root-n in the `stderr` property, it is not possible to get valid standard errors for MSR Banzhaf.